Fails to build

```
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/uset.h:1763:5: error: 'UnicodeString' does not name a type
 1763 |     UnicodeString operator*() const {
      |     ^~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/uset.h: In member function 'icu_76::header::USetElementIterator& icu_76::header::USetElementIterator::operator++()':
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/uset.h:1785:66: error: 'nullptr' was not declared in this scope
 1785 |             int32_t result = uset_getItem(uset, index, &c, &end, nullptr, 0, &errorCode);
      |                                                                  ^~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h: At global scope:
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1553:25: error: 'enable_if_t' in namespace 'std' does not name a template type
 1553 |         typename = std::enable_if_t<ConvertibleToU16StringView<T> && ConvertibleToU16StringView<U>>>
      |                         ^~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1553:20: note: 'std::enable_if_t' is only available from C++14 onwards
 1553 |         typename = std::enable_if_t<ConvertibleToU16StringView<T> && ConvertibleToU16StringView<U>>>
      |                    ^~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1553:36: error: expected '>' before '<' token
 1553 |         typename = std::enable_if_t<ConvertibleToU16StringView<T> && ConvertibleToU16StringView<U>>>
      |                                    ^
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1554:55: error: expected unqualified-id before '{' token
 1554 |     bool operator()(const T& lhs, const U& rhs) const {
      |                                                       ^
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1559:26: error: 'std::string_view' has not been declared
 1559 |     bool operator()(std::string_view lhs, std::string_view rhs) const {
      |                          ^~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1559:48: error: 'std::string_view' has not been declared
 1559 |     bool operator()(std::string_view lhs, std::string_view rhs) const {
      |                                                ^~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1571:16: error: 'UnicodeString' has not been declared
 1571 |     bool match(UnicodeString lhs, UnicodeString rhs) const {
      |                ^~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1571:35: error: 'UnicodeString' has not been declared
 1571 |     bool match(UnicodeString lhs, UnicodeString rhs) const {
      |                                   ^~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1580:16: error: 'StringPiece' has not been declared
 1580 |     bool match(StringPiece lhs, StringPiece rhs) const {
      |                ^~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1580:33: error: 'StringPiece' has not been declared
 1580 |     bool match(StringPiece lhs, StringPiece rhs) const {
      |                                 ^~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1580:10: error: 'bool icu_76::header::collator::internal::Predicate<Compare, result>::match(int, int) const' cannot be overloaded with 'bool icu_76::header::collator::internal::Predicate<Compare, result>::match(int, int) const'
 1580 |     bool match(StringPiece lhs, StringPiece rhs) const {
      |          ^~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1571:10: note: previous declaration 'bool icu_76::header::collator::internal::Predicate<Compare, result>::match(int, int) const'
 1571 |     bool match(UnicodeString lhs, UnicodeString rhs) const {
      |          ^~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1592:12: error: 'constexpr' does not name a type
 1592 |     static constexpr Compare<UCollationResult> compare{};
      |            ^~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1592:12: note: C++11 'constexpr' only available with '-std=c++11' or '-std=gnu++11'
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h: In member function 'bool icu_76::header::collator::internal::Predicate<Compare, result>::match(int, int) const':
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1575:32: error: request for member 'getBuffer' in 'lhs', which is of non-class type 'int'
 1575 |                 toUCharPtr(lhs.getBuffer()), lhs.length(),
      |                                ^~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1575:17: error: there are no arguments to 'toUCharPtr' that depend on a template parameter, so a declaration of 'toUCharPtr' must be available [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-fpermissive-fpermissive8;;]
 1575 |                 toUCharPtr(lhs.getBuffer()), lhs.length(),
      |                 ^~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1575:17: note: (if you use '8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-fpermissive-fpermissive8;;', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1575:50: error: request for member 'length' in 'lhs', which is of non-class type 'int'
 1575 |                 toUCharPtr(lhs.getBuffer()), lhs.length(),
      |                                                  ^~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1576:32: error: request for member 'getBuffer' in 'rhs', which is of non-class type 'int'
 1576 |                 toUCharPtr(rhs.getBuffer()), rhs.length()),
      |                                ^~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1576:17: error: there are no arguments to 'toUCharPtr' that depend on a template parameter, so a declaration of 'toUCharPtr' must be available [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-fpermissive-fpermissive8;;]
 1576 |                 toUCharPtr(rhs.getBuffer()), rhs.length()),
      |                 ^~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1576:50: error: request for member 'length' in 'rhs', which is of non-class type 'int'
 1576 |                 toUCharPtr(rhs.getBuffer()), rhs.length()),
      |                                                  ^~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h: In member function 'bool icu_76::header::collator::internal::Predicate<Compare, result>::match(int, int) const':
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1585:21: error: request for member 'data' in 'lhs', which is of non-class type 'int'
 1585 |                 lhs.data(), lhs.length(),
      |                     ^~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1585:33: error: request for member 'length' in 'lhs', which is of non-class type 'int'
 1585 |                 lhs.data(), lhs.length(),
      |                                 ^~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1586:21: error: request for member 'data' in 'rhs', which is of non-class type 'int'
 1586 |                 rhs.data(), rhs.length(),
      |                     ^~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1586:33: error: request for member 'length' in 'rhs', which is of non-class type 'int'
 1586 |                 rhs.data(), rhs.length(),
      |                                 ^~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1582:16: error: there are no arguments to 'compare' that depend on a template parameter, so a declaration of 'compare' must be available [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-fpermissive-fpermissive8;;]
 1582 |         return compare(
      |                ^~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h: At global scope:
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1602:7: error: expected nested-name-specifier before 'equal_to'
 1602 | using equal_to = internal::Predicate<std::equal_to, UCOL_EQUAL>;
      |       ^~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1609:7: error: expected nested-name-specifier before 'greater'
 1609 | using greater = internal::Predicate<std::equal_to, UCOL_GREATER>;
      |       ^~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1616:7: error: expected nested-name-specifier before 'less'
 1616 | using less = internal::Predicate<std::equal_to, UCOL_LESS>;
      |       ^~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1623:7: error: expected nested-name-specifier before 'not_equal_to'
 1623 | using not_equal_to = internal::Predicate<std::not_equal_to, UCOL_EQUAL>;
      |       ^~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1630:7: error: expected nested-name-specifier before 'greater_equal'
 1630 | using greater_equal = internal::Predicate<std::not_equal_to, UCOL_LESS>;
      |       ^~~~~~~~~~~~~
/nix/store/x4mxg72g9k7srfvp7pndgn7r92savaf6-icu4c-76.1-dev/include/unicode/ucol.h:1637:7: error: expected nested-name-specifier before 'less_equal'
 1637 | using less_equal = internal::Predicate<std::not_equal_to, UCOL_GREATER>;
      |       ^~~~~~~~~~
make[3]: *** [make.rules:106: /build/source/temp/Release/common/unicode_util.o] Error 1
make[3]: Leaving directory '/build/source/gen'
make[2]: *** [Makefile:192: master_process] Error 2
make[2]: Leaving directory '/build/source/gen'
make[1]: *** [Makefile:66: firebird] Error 2
make[1]: Leaving directory '/build/source/gen'
make: *** [Makefile:6: firebird] Error 2
error: builder for '/nix/store/yvsk8b8m0053hfa0b3ggg4dm78r6a6v3-firebird-3.0.12.drv' failed with exit code 2;
```

Maybe needs an icu version change?

--- a/pkgs/servers/firebird/default.nix
+++ b/pkgs/servers/firebird/default.nix
@@ -10,7 +10,7 @@ let base = {
     changelog = "https://github.com/FirebirdSQL/firebird/blob/master/CHANGELOG.md";
     license = [ "IDPL" "Interbase-1.0" ];
     platforms = platforms.linux;
-    maintainers = with maintainers; [ marcweber ];
+    maintainers = with maintainers; [ bbenno marcweber ];
   };
 
   nativeBuildInputs = [ autoreconfHook271 ];

--- a/pkgs/servers/firebird/default.nix
+++ b/pkgs/servers/firebird/default.nix
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub, libedit, autoreconfHook271, zlib, unzip, libtommath, libtomcrypt, icu, superServer ? false }:
+{ lib, stdenv, fetchFromGitHub, libedit, autoreconfHook271, zlib, unzip, libtommath, libtomcrypt, icu73, superServer ? false }:
 
 let base = {
   pname = "firebird";
@@ -15,9 +15,9 @@ let base = {
 
   nativeBuildInputs = [ autoreconfHook271 ];
 
-  buildInputs = [ libedit icu ];
+  buildInputs = [ libedit icu73 ];
 
-  LD_LIBRARY_PATH = lib.makeLibraryPath [ icu ];
+  LD_LIBRARY_PATH = lib.makeLibraryPath [ icu73 ];
 
   configureFlags = [
     "--with-system-editline"
@@ -55,13 +55,13 @@ let base = {
   });
 
   firebird_3 = stdenv.mkDerivation (base // rec {
-    version = "3.0.10";
+    version = "3.0.12";
 
     src = fetchFromGitHub {
       owner = "FirebirdSQL";
       repo = "firebird";
       rev = "v${version}";
-      sha256 = "sha256-PT2b3989n/7xLGNREWinEey9SGnAXShITdum+yiFlHY=";
+      hash = "sha256-po8tMrOahfwayVXa7Eadr9+ZEmZizHlCmxi094cOJSY=";
     };
 
     buildInputs = base.buildInputs ++ [ zlib libtommath ];

